src/rpdk/core/contract/suite/handler_update.py

```diff
@@ -26,6 +27,11 @@ def updated_resource(resource_client):
         updated_input_model = update_request = resource_client.generate_update_example(
             created_model
         )
+
+        updated_input_model = prune_properties_from_model(
+            updated_input_model, resource_client.read_only_paths
```

Comment:
I like it. Some resources use the readOnlyProperties as to support `GetAtt`  and mixed the readOnly with resource properties. This one can also help to detection such mis-use.